compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt

```diff
@@ -486,6 +486,11 @@ class BodyResolveContext(
         return withTowerDataCleanup {
             addLocalScope(FirLocalScope())
             if (function is FirSimpleFunction) {
```

Comment:
What about `FirConstructor`?

Reply:
Will be handled at `forConstructorBody` 20-ish lines below: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt#L498

There, all parameters are added to the scope: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt#L509

This one seems related too: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt#L450

Reply:
But it does not seem to work for non-primary constructors. They still appeared to be unresolved.

Reply:
How I didn't spot this! 🤦  https://github.com/JetBrains/kotlin/pull/4321/files#diff-785bbdf0d10bf7bf27c298b2d1bcffab38fb02e72cc1240b1b5450ff0ed3fb86L10

Thank you for keeping asking this!